### ISSUE TITLE
CMake and QTCreator integration

### ISSUE THREAD

**Issue by ochagnon** [OlivierLDff/Qaterial#78]
Warning: I'm not familiar with `cmake`, my experience had been with `qmake` so far. My question will probably be asked again and again, as QT Creator is a popular IDE for Qt/QML applications.

I'm trying to use Qaterial with QTCreator. In this [Github repo](https://github.com/ochagnon/qaterial-qtcreator-bug), I provide the code I use to reproduce this problem, with 1 commit per change I perform. You can very easily try yourself.

So, I start from scratch and create a [`Test` project that use `cmake`](https://github.com/ochagnon/qaterial-qtcreator-bug/commit/769b4e6a484dfefa3d1ab7d8c9bd2e3993c127e5). This commit just the initial output of Qt Creator.

Then, I add [this commit](https://github.com/ochagnon/qaterial-qtcreator-bug/commit/e9657acd561f901143156a9710276aa296831fa3) that switch C++ and QML to Qaterial and add dependency build.

But, that don't works, build fail with 

```console
Test/main.cpp:1:10: fatal error: 'Qaterial/Qaterial.hpp' file not found
#include <Qaterial/Qaterial.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Sure, the reason is those files are in  `build-Test-Desktop_Qt_5_15_1_clang_64bit-Debug/_deps/qaterial-src/`.

From [QtCreator cmake integration doc](https://doc.qt.io/qtcreator/creator-project-cmake.html):

> Adding External Libraries to CMake Projects
> Through external libraries, Qt Creator can support code completion and syntax highlighting as if they were part of the current project or the Qt library.
> Qt Creator detects the external libraries using the find_package() macro. Some libraries come with the CMake installation. You can find those in the Modules directory of your CMake installation. For more information, see cmake-packages(7).
> Syntax completion and highlighting work once your project successfully builds and links against the external library.

So, I tried `find_package(Qaterial)` and  output of cmake is:

```console
Make Warning at CMakeLists.txt:38 (find_package):
  By not providing "FindQaterial.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qaterial",
  but CMake did not find one.

  Could not find a package configuration file provided by "Qaterial" with any
  of the following names:

    QaterialConfig.cmake
    qaterial-config.cmake

  Add the installation prefix of "Qaterial" to CMAKE_PREFIX_PATH or set
  "Qaterial_DIR" to a directory containing one of the above files.  If
  "Qaterial" provides a separate development package or SDK, be sure it has
  been installed.
```

In Qaterial repository, there is no `*config*.cmake` nor `Find*.cmake`.

So, how can I use Qaterial with Qt Creator?

**Comment by ochagnon** [comment]
I think I found how

**Comment by ochagnon** [comment]
Fixed in https://github.com/ochagnon/qaterial-qtcreator-bug/commit/40dc0f5ff2d37dcae7580510554ce7445c57ba55

**Comment by ochagnon** [comment]
So, the binary start and can load QML, but the UI of Qt creator still complains that `QML module not found (Qaterial)`

**Comment by OlivierLDff** [comment]
Ok, so for now all your problems are not related to qt creator but to cmake (which are the one everyone is struggling with starting with cmake)
I saw your code and I Will investigate. It is strange since you added the addImportPath

**Comment by ochagnon** [comment]
@OlivierLDff thanks, [this doc](https://doc.qt.io/qtcreator/creator-qml-modules-with-plugins.html) give a few possible fixes for the missing qml in creator itself, I'm also investigating

**Comment by OlivierLDff** [comment]
You should use https://github.com/OlivierLDff/QaterialGallery as a reference

**Comment by ochagnon** [comment]
I fixed QT Creator in [this commit](https://github.com/ochagnon/qaterial-qtcreator-bug/commit/18c4afd4fdb0793cdd25bf6f21cb45c9c0039e44)

**Comment by OlivierLDff** [comment]
This kind of ugly fix, because it is hard coded path. When deploying your app it will be a nightmare.
Anyway, qaterial's qml files should be embedded in qrc/Qaterial. And there should be a qrc/Qaterial/qmldir in there.

**Comment by ochagnon** [comment]
> This kind of fix is ugly, because it is hard coded path. When deploying your app it will be a nightmare.

that don't solve everything, there is additional issues

`module: "Qaterial.Impl.Icons" is not installed`

**Comment by ochagnon** [comment]
> Anyway, qaterial's qml files should be embedded in qrc/Qaterial. And there should be a qrc/Qaterial/qmldir in there.

how?

**Comment by ochagnon** [comment]
> Anyway, qaterial's qml files should be embedded in qrc/Qaterial. And there should be a qrc/Qaterial/qmldir in there.

I'm looking at [gallery as a reference](https://github.com/OlivierLDff/QaterialGallery/blob/master/CMakeLists.txt) and I don't see an obvious command that embed Qaterial into the gallery

**Comment by OlivierLDff** [comment]
Yes because you are missing a file in qrc:/Qaterial/Impl/Icons which is only in qrc (because generated)

**Comment by OlivierLDff** [comment]
I opened your example on windows
```
git clone https://github.com/ochagnon/qaterial-qtcreator-bug
cd qaterial-qtcreator-bug
mkdir build
cd build/
cmake ..
cmake --build . -j
```

And the app launched.

Did you clean your cmake build folder?

**Comment by ochagnon** [comment]
> Yes because you are missing a file in qrc:/Qaterial/Impl/Icons which is only in qrc (because generated)

But, how can I add this resource with mine? 

I don't see anything in [this directory](https://github.com/OlivierLDff/QaterialGallery/tree/master/qml)
There is no additional `.qrc` loaded in [Main.cpp](https://github.com/OlivierLDff/QaterialGallery/blob/master/src/Main.cpp)
I don't get any cmake build steps that combine project qml with the Qaterial dependencies :(

**Comment by OlivierLDff** [comment]
You shouldn't have anything to do, that is why it is strange.
![image](https://user-images.githubusercontent.com/17255804/96412073-d1cf2d80-11e9-11eb-9f63-ad9b8f5ac542.png)
For me it worked without any problem.
You should try to `rm -rf *` your build folder and do `cmake ..` again.

**Comment by OlivierLDff** [comment]
![demo qr creator](https://user-images.githubusercontent.com/17255804/96412663-c3354600-11ea-11eb-854d-43e8d1194ce9.gif)

**Comment by ochagnon** [comment]
> And the app launched.
> Did you clean your cmake build folder?

yes, the application build and run, but the problem is when I try to open QtCreator designer (very important for development).

First, I had to add `import QtQuick 2.12` to my `Main.qml` and:

![Screen Shot 2020-10-19 at 15 14 56](https://user-images.githubusercontent.com/1025638/96413186-e2999680-121d-11eb-827b-3b562d4aaae2.png)

This is where the missing icons error show up. This is probably because I don't use any icons at this point in my `Main.qml`. The very minimalistic things I tried to do don't trigger that error.

I tried with Gallery demo app, and I have an other problem.

![Screen Shot 2020-10-19 at 15 04 09](https://user-images.githubusercontent.com/1025638/96413198-e9280e00-121d-11eb-9de8-ed1716cbc17f.png)

**Comment by ochagnon** [comment]
The problem is only in QtCreator in "Design" mode

**Comment by OlivierLDff** [comment]
I never used design, I know qaterial should be compiled as a plugin for it to work and there is special requirements.
Qaterial isn't only qml files.
Sorry I don't have time to support that on my free time since I'm not using it but feel free to create a pull request that address the issue

**Comment by OlivierLDff** [comment]
The file you are missing are located in your build folder `res/Icons/Generated` & `res/Fonts/Generated`

**Comment by alvaroyecla** [comment]
@ochagnon  when compiled either plugin must be added in importPaths in .qmlproject or C:\Qt\Tools\QtDesignStudio\qt6_design_studio_reduced_version\qml (thats where i have  it installed)

You have to let Qt Desing Studio know where the plugin is
